tools/cli/commands/changelog.js

```diff
@@ -390,16 +412,59 @@ async function changelogArgs( argv ) {
 				argv.args.push( argv.ver );
 			}
 			break;
+		case 'squash':
+			if ( argv.r && argv.readme ) {
+				readme = true;
+			} else {
+				readme = await promptReadme( argv );
+			}
```

Comment:
This should let `--no-readme` bypass the prompt.
```suggestion
			if ( typeof argv.readme === 'undefined' ) {
				readme = await promptReadme( argv );
			} else {
				readme = argv.readme;
			}
```